src/tests/testCbcPointer.cc

```diff
@@ -0,0 +1,260 @@
+        CPPUNIT_ASSERT(copyable.valid() != nullptr);
+        // clear() must not deallocate
+        copyable.clear();
+        CPPUNIT_ASSERT(copyable.valid() == nullptr);
+        CPPUNIT_ASSERT(moveable.valid() != nullptr);
+        CPPUNIT_ASSERT_EQUAL(1, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(false, !moveable); // still valid
+
+        // delete with 2 active references should deallocate
+        delete one.raw();
+        CPPUNIT_ASSERT_EQUAL(0, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(true, !moveable); // now invalid
+    }
+
+    {
+        const CbcPointer<Blob> one(new Blob);
+        const CbcPointer<Blob> two(new Blob);
+        CPPUNIT_ASSERT(one.valid() != nullptr);
+        CPPUNIT_ASSERT(two.valid() != nullptr);
+        CPPUNIT_ASSERT(one.raw() != two.raw()); // paranoia
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+
+        // copy should not allocate
+        CbcPointer<Blob> copyable(one);
+        CPPUNIT_ASSERT(copyable.valid() != nullptr);
+        CPPUNIT_ASSERT_EQUAL(one, copyable);
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+        copyable = two;
+        CPPUNIT_ASSERT(copyable.valid() != nullptr);
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+        CPPUNIT_ASSERT(copyable.raw() != one.raw());
+        CPPUNIT_ASSERT_EQUAL(two, copyable);
+        copyable = copyable;
+        CPPUNIT_ASSERT(copyable.valid() != nullptr);
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(two, copyable);
+
+        // move should not allocate
+        CbcPointer<Blob> moveable(std::move(copyable));
+        CPPUNIT_ASSERT(copyable.valid() == nullptr); // moved away
+        CPPUNIT_ASSERT(moveable.valid() != nullptr);
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(two, moveable);
+        copyable = one; // reset for next test
+        moveable = std::move(copyable);
+        CPPUNIT_ASSERT(copyable.valid() == nullptr); // moved away
+        CPPUNIT_ASSERT(moveable.valid() != nullptr);
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(one, moveable);
+        moveable = std::move(moveable);
+        CPPUNIT_ASSERT(moveable.valid() != nullptr); // no change
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(one, moveable);
+
+        copyable = one; // reset for next test
+        CPPUNIT_ASSERT(copyable.valid() != nullptr);
+        // clear() must not deallocate
+        copyable.clear();
+        CPPUNIT_ASSERT(copyable.valid() == nullptr);
+        CPPUNIT_ASSERT(moveable.valid() != nullptr);
+        CPPUNIT_ASSERT_EQUAL(2, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(false, !moveable); // still valid
+
+        // delete with 2 active references should deallocate
+        CPPUNIT_ASSERT_EQUAL(one, moveable);
+        delete one.raw();
+        CPPUNIT_ASSERT_EQUAL(1, Blob::Instances);
+        CPPUNIT_ASSERT_EQUAL(true, !moveable); // now invalid
```

Comment:
Since we are not testing the negation operator here, let's call valid() directly:

```suggestion
        CPPUNIT_ASSERT_EQUAL(false, moveable.valid());
```

If you do want to add a negation operator test, we can (especially when this code is moved to a method that is not specific to so called "counting" checks as discussed in another change request):

```suggestion
        CPPUNIT_ASSERT_EQUAL(false, moveable.valid());
        CPPUNIT_ASSERT_EQUAL(true, !moveable);
```

Reply:
You misunderstand, this *is* testing the negation operator.

Reply:
> this _is_ testing the negation operator.

If this code is testing the negation operator, then please remove the comment that implies that it is testing cbdata validity. The two concepts are distinct/different and the comment may mislead new developers into thinking that they are equivalent.

```suggestion
        CPPUNIT_ASSERT_EQUAL(true, !moveable);
```

or, simply

```suggestion
        CPPUNIT_ASSERT(!moveable);
```

However, I think it is _better_ to test validity in this context (or both validity and negation if you prefer). The original change request contains the corresponding suggestions.

----

> You misunderstand, ...

IMHO, whenever the reviewer misunderstood something, the author should ask themselves: How do I adjust my PR to avoid similar misunderstandings by future code/commit readers? Sometimes, the reviewer just made a mistake that others are very unlikely to repeat or that cannot be prevented with in-scope PR changes. However, in the vast majority of cases you are dealing with, a PR adjustment _is_ warranted, while simply explaining what your code means to you, via a GitHub comment, is not going to address the essence of the change request (and, hence, will necessitate one more review iteration).